src/Kentico.Xperience.Siteimprove/Services/SiteimproveService.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            if (string.IsNullOrEmpty(token))
+            {
+                token = await GetTokenInternal(cancellationToken);
+            }
+
+            return token;
+        }
+
+
+        /// <inheritdoc/>
+        public async Task CheckPages(IEnumerable<string> urls, CancellationToken? cancellationToken = null)
+        {
+            var client = HttpClient;
+
+            foreach (var url in urls)
+            {
+                string requestPath = string.Format(SiteimproveConstants.PAGE_CHECK_PATH, siteID, url);
+                var response = await Post(requestPath, cancellationToken, client);
+
+                if (response != null && !response.IsSuccessStatusCode)
+                {
+                    eventLogService.LogError(nameof(SiteimproveService), nameof(CheckPages), $"Failed to request check of a page '{url}'.");
+                }
+            }
+        }
+
+
+        /// <inheritdoc/>
+        public Task<bool> IsContentCheckEnabled(CancellationToken? cancellationToken = null)
+        {
+            return Task.FromResult(isContentCheckEnabled);
+        }
+
+
+        /// <inheritdoc/>
+        public async Task Initialize(CancellationToken? cancellationToken = null)
+        {
+            isContentCheckEnabled = await IsContentCheckEnabledInternal(cancellationToken);
+            siteID = await GetSiteID(cancellationToken);
+        }
+
+
+        internal async Task<long> GetSiteID(CancellationToken? cancellationToken = null)
+        {
+            var client = HttpClient;
+            string requestPath = string.Format(SiteimproveConstants.SITES_PATH, 1);
+            var sites = await Get<SiteimproveSites>(requestPath, cancellationToken, client);
+
+            if (sites == null)
+            {
+                return siteID;
+            }
+
+            requestPath = string.Format(SiteimproveConstants.SITES_PATH, sites.TotalItems);
+
+            sites = await Get<SiteimproveSites>(requestPath, cancellationToken, client);
+
+            if (sites == null || sites.Sites == null)
+            {
+                return siteID;
+            }
+
+            string domain = GetDomain();
+
+            var site = sites.Sites.FirstOrDefault(s => s.SiteUrl != null && s.SiteUrl.Contains($"://{domain}"));
+
+            if (site == null)
+            {
+                eventLogService.LogError(nameof(SiteimproveService), nameof(GetSiteID), $"Site with URL '{domain}' could not be found on Siteimprove.");
+                return siteID;
+            }
+
+            return site.SiteID;
+        }
+
+
+        internal async Task<bool> IsContentCheckEnabledInternal(CancellationToken? cancellationToken = null)
+        {
+            if (!siteimproveOptions.EnableContentCheck)
+            {
+                return false;
+            }
+
+            var client = HttpClient;
+            bool isContentCheckEnabled = await IsContentCheckReady(client, cancellationToken);

```

Comment:
There is already a class field with the same name. Was it intentional to hide that field with this variable?

Reply:
Yes, it made more sense to me to have the method return the value rather than modifying the field.